demo/README.md

```diff
@@ -0,0 +1,31 @@
+## Demo
+
+This demo deploys the Documentation section with the Nuxt-Security module protection.
+
+### Deploying on Vercel
+
+Setup your Vercel Project Settings depending on whether you want an SSR deployment or an SSG deployment:
+
+#### SSR
+
+1. Set **Root Directory** field to `demo`
+
+![Vercel SSR example](vercel-ssr.png "Set the Root Directory")
+
+
+#### SSG
+
+1. Set **Root Directory** field to `demo`
+2. Override **Build Command** field to `yarn generate`
+
+![Vercel SSG example](vercel-ssg.png "Set the Root Directory and the Build Command")
+
+### Pre-deployed Examples
+
+You can navigate to the following links to see it live in action 🚀
+
+https://secure-docs-ssr.vercel.app/
```

Comment:
question: would you like me to deploy these documentation pages on my Vercel account? Or will they be updated after I merge the code to the repo?

Reply:
They won't be updated because I set up my own Vercel configuration so that it only deploys from the demo-docs-security branch.
I think it's better if you own these pages anyway.
I just cancelled these domain names from my account so that you can take over them.

Reply:
Ok, maybe I will also change their names to for example `nuxt-security-ssr` and `nuxt-security-ssg`?

Reply:
I have also been thinking about maybe adding info about these demo apps to the documentation itself? 

So that user can easily check both of them and how they are deployed to production.

What do you think?

Reply:
BTW feel free to not merge this PR if you think it's polluting the overall codebase.
I thought about your comments and I think you're right. After all, we can keep this as a separate branch for us internally to make tests.

Reply:
> I have also been thinking about maybe adding info about these demo apps to the documentation itself?
> 
> So that user can easily check both of them and how they are deployed to production.
> 
> What do you think?

Yes ok, like a new section in the docs 'Demo' or 'Check it live' that tells the user to click on the links and open their devtools to observe headers?

Reply:
Although the more I think about it, isn't that going to confuse the average user ?
The guy is scrolling the docs and we redirect him on a clone of the docs with nuxt-security 🤯
That's why your idea of having a specific demo app (e.g. movies) is maybe cleaner? That's a lot of work though, unless we can fork some existing example. Let me know

Reply:
ok now I think I might have an idea of how to make this easily
1. We fork Nuxt Movies
2. We modify the fork to make it an extendable theme
3. We modify `demo` to import that fork instead of importing `docs`

That's it. This way we can both
- play locally with demo during dev based on the current state of nuxt-security in dev
- deploy a demo based on Nuxt Movies as a showcase to the users with links in the docs, but based on the latest released version.
WDYT? Is it worth the effort you think?

Reply:
I would say that it is a much better idea! 

It also showcases another Nuxt example project that covers good practices and patterns. 

There are two example apps that we should be able to easily showcase as demos:

1. https://github.com/nuxt/hackernews
2. https://github.com/nuxt/movies

i think this could make a better example/demo showcase of how the user could setup their application (whether it is a SSG or SSR) based on our recommendations.

I would recommend creating a folder called `demo` and inside of it have two folders for both movies and hackernews applications named `ssr` & `ssg`. They could be then deployed from my Vercel account with appriopriate names like (nuxt-security-ssr and nuxt-security-ssg) and then, we could easily link them in our documentation and readme for both users and us to test out :)

But I wouldn't spend that much time on developing these applications. Just take one or two, add nuxt-security into them and push them to this or new branch so that I could deploy them and link in the docs. This should be enough for both us and users to test NuxtSecurity in battle :)

Let me know what you think about it.